File: 198-house-robber/198-house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        vector<int> next(2, 0), cur(2, 0);
        
        
        
        for(int i = n-1; i >= 0; i--) {
            int pick = 0;
            int notPick = 0;
            pick = nums[i] + next[false];
            notPick = next[true];
            
            cur[true] = max(pick, notPick);
            cur[false] = notPick;
            
            next = cur;
            
        }
        
        return next[true];
    }
};

File: 198-house-robber/198-house-robber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {
    const n = nums.length
    let nex = 0, nex2 = 0
    
    for(let i = n-1; i >= 0; i--) {
        let pick = nums[i] + nex2
        let notPick = nex
        // console.log(pick, notPick)
        nex2 = nex
        nex = (pick < notPick) ? notPick : pick
    }
    
    return nex
};